src/html_reporter.cc

```diff
@@ -0,0 +1,507 @@
+    "                    },\n"
+    "                    title: {\n"
+    "                        text: '${BENCHMARK_NAME}'\n"
+    "                    },\n"
+    "                    legend: {\n"
+    "                         align: 'right',\n"
+    "                         verticalAlign: 'top',\n"
+    "                         layout: 'vertical'\n"
+    "                    },\n"
+    "                    xAxis: {\n"
+    "                        title: {\n"
+    "                              text: 'Argument 1',\n"
+    "                              enabled: true\n"
+    "                        },\n"
+    "                        allowDecimals: true\n"
+    "                    },\n"
+    "                    yAxis: {\n"
+    "                        title: {\n"
+    "                            text: '${YAXIS_TITLE}'\n"
+    "                        },\n  "
+    "                        allowDecimals: true\n"
+    "                    },\n"
+    "                    tooltip: {\n"
+    "                        headerFormat: '${TOOLTIP_HEADER}',\n"
+    "                        pointFormat: '${TOOLTIP_BODY}',\n"
+    "                        borderWidth: 5\n"
+    "                    },\n"
+    "                    scrollbar: {\n"
+    "                        enabled: true\n"
+    "                    },\n"
+    "                    navigator: {\n"
+    "                        enabled: true\n"
+    "                    },\n"
+    "                    plotOptions: {\n"
+    "                        line: {\n"
+    "                          marker: {\n"
+    "                              enabled: true,\n"
+    "                              radius: 3\n"
+    "                          },\n"
+    "                          lineWidth: 1\n"
+    "                        }\n"
+    "                    },\n"
+    "                   series:[\n${SERIES}"
+    "                   ]\n"
+    "                });\n${CHART}";
+
+static const char *const div_Element =
+    "      <div class = \"chart\" id = \"${DIV_NAME}\"></div>\n${DIV}";
+
+static const char *const HTML_Base =
+    "<!DOCTYPE html>\n"
+    "<html>\n"
+    "   <head>\n"
+    "       <meta charset = \"UTF-8\"/>\n"
+    "       <title>Benchmark</title>\n"
+    "       <style type = \"text/css\">\n"
+    "           .chart{\n"
+    "                height:400px;\n"
+    "           }\n"
+    "       </style>\n"
+    "@CMAKE_JAVASCRIPT_REPLACEMENT@"
+    "       <script>\n"
+    "            $(function () {\n"
+    "${CHART}"
+    "            });\n"
+    "       </script>\n"
+    "   </head>\n"
+    "   <body>\n"
+    "${DIV}"
+    "   </body>\n"
+    "</html>";
+
+namespace benchmark {
+std::string &replaceString(std::string &input, const std::string &from,
+                           const std::string &to) {
+  size_t position = input.find(from);
+
+  if (position != std::string::npos) {
+    input.replace(position, from.size(), to);
+  }
+
+  return input;
```

Comment:
why return input and replace inline? either take input as const & and take a copy, or just do the inline replacement. however, if you do the inline replacement, input should be a *.

https://google.github.io/styleguide/cppguide.html#Reference_Arguments